tests/unit/test_app.py

```diff
@@ -0,0 +1,252 @@
+          'CreationTime': datetime(2021, 7, 11, tzinfo=timezone.utc),
+          'DeletionTime': datetime(2021, 8, 21, tzinfo=timezone.utc),
+          'StackStatus': 'DELETE_COMPLETE',
+          'DriftInformation': {
+            'StackDriftStatus': 'NOT_CHECKED'
+          }
+        }
+      ]
+    }
+
+    list_stack_resources_response = {
+      'StackResourceSummaries': [
+        {
+          'LogicalResourceId': 'S3Bucket',
+          'PhysicalResourceId': 'sc-465877038949-pp-qzha7mc257fbo-s3bucket-1m9wbdmmcjukq',
+          'ResourceType': 'AWS::S3::Bucket',
+          'LastUpdatedTimestamp': datetime(2021, 6, 28, tzinfo=timezone.utc),
+          'ResourceStatus': 'DELETE_SKIPPED',
+          'DriftInformation': {
+            'StackResourceDriftStatus': 'NOT_CHECKED'
+          }
+        },
+        {
+          'LogicalResourceId': 'SynapseOwnerFile',
+          'PhysicalResourceId': 's3://sc-465877038949-pp-qzha7mc257fbo-s3bucket-1m9wbdmmcjukq/owner.txt',
+          'ResourceType': 'Custom::S3Object',
+          'LastUpdatedTimestamp': datetime(2021, 6, 28, tzinfo=timezone.utc),
+          'ResourceStatus': 'DELETE_COMPLETE',
+          'DriftInformation': {
+            'StackResourceDriftStatus': 'NOT_CHECKED'
+          }
+        }
+      ]
+    }
+
+    purge_date_mock.return_value = datetime(2021, 9, 12, tzinfo=timezone.utc)
+    get_stacks_mock.return_value = list_deleted_stacks_response['StackSummaries']
+    get_stack_resources_mock.return_value = list_stack_resources_response['StackResourceSummaries']
+
+    sc_stacks = app._get_deleted_stacks(30)
+    self.assertEqual(1, len((sc_stacks)))
+    self.assertEqual(sc_stacks[0]['StackName'], "SC-465877038949-pp-qzha7mc257fbo")
+    self.assertEqual(sc_stacks[0]['AssociatedResource'],
+                     "sc-465877038949-pp-qzha7mc257fbo-s3bucket-1m9wbdmmcjukq")
+
+
+  @patch('sc_bucket_cleanup.app._get_deleted_stacks')
+  @patch('sc_bucket_cleanup.app._get_buckets')
+  @patch('sc_bucket_cleanup.app._delete_bucket')
+  def test_lambda_handler_cleanup_no_buckets_no_stacks(self, delete_bucket_mock,
+                                    get_buckets_mock, get_deleted_stacks_mock):
+    """
+    This tests that AWS list stacks returns no stacks
+    """
+
+    get_deleted_stacks_mock.return_value = []
+    get_buckets_mock.return_value = self.list_buckets_response['Buckets']
+    delete_bucket_mock.return_value = ""
+    app.lambda_handler("","")
+    self.assertEqual(delete_bucket_mock.call_count, 0)
+
+
+  @patch('sc_bucket_cleanup.app._get_deleted_stacks')
+  @patch('sc_bucket_cleanup.app._get_buckets')
+  @patch('sc_bucket_cleanup.app._delete_bucket')
+  def test_lambda_handler_cleanup_no_buckets_no_matching_buckets(self, delete_bucket_mock,
+                                    get_buckets_mock, get_deleted_stacks_mock):
+    """
+    This tests that AWS list stacks returns no stacks
+    """
+    get_deleted_stacks_response = {
+      'StackSummaries': [
```

Comment:
Can you add a comment explaining why this stack isn't deleted?

Reply:
sorry, the comments are from me doing copy and paste then i forgot to update the comment.  I have updated them so should be more clear now.